pkgs/by-name/ea/easyeffects/package.nix

```
@@ -1,53 +1,59 @@
-{ lib
-, stdenv
-, appstream-glib
-, desktop-file-utils
-, deepfilternet
-, fetchFromGitHub
-, calf
-, fftw
-, fftwFloat
-, fmt_9
-, glib
-, gsl
-, gtk4
-, itstool
-, ladspaH
-, libadwaita
-, libbs2b
-, libebur128
-, libportal-gtk4
-, libsamplerate
-, libsigcxx30
-, libsndfile
-, lilv
-, lsp-plugins
-, lv2
-, mda_lv2
-, meson
-, ninja
-, nlohmann_json
-, pipewire
-, pkg-config
-, rnnoise
-, rubberband
-, speexdsp
-, soundtouch
-, tbb
-, wrapGAppsHook4
-, zam-plugins
-, zita-convolver
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  appstream-glib,
+  calf,
+  deepfilternet,
+  desktop-file-utils,
+  fftw,
+  fftwFloat,
+  fmt_9,
+  glib,
+  gsl,
+  gtk4,
+  itstool,
+  ladspaH,
+  libadwaita,
+  libbs2b,
+  libebur128,
+  libportal-gtk4,
+  libsamplerate,
+  libsigcxx30,
+  libsndfile,
+  lilv,
+  lsp-plugins,
+  lv2,
+  mda_lv2,
+  meson,
+  ninja,
+  nix-update-script,
+  nlohmann_json,
+  pipewire,
+  pkg-config,
+  rnnoise,
+  rubberband,
+  speexdsp,
+  soundtouch,
+  tbb,
+  wrapGAppsHook4,
+  zam-plugins,
+  zita-convolver,
 }:
+let
+  # Fix crashes with speexdsp effects
+  speexdsp' = speexdsp.override { withFftw3 = false; };
+in
 
 stdenv.mkDerivation rec {
```

Prefer using something like this instead of `rec`:

```suggestion
stdenv.mkDerivation (finalAttrs: {
```

---

the use of `finalAttrs` over `rec` would be purely cosmetic as no recursion is required. i don't see a point in adopting this until it is, especially when it can potentially add performance overhead

---

If you don't want to use `finalAttrs`, you can also do something like:

```nix
let
  version = "7.1.7";
in
stdenv.mkDerivation {
  pname = "easyeffects";
  inherit version;
  # ...
```

if you want to reuse that later for the changelog. My point is just that `rec` over an entire `stdenv.mkDerivation`'s attributes makes it easy to lose track of what is already defined.

---

this is again, a cosmetic change. given how simple (and few) attributes are at the top-level, this will be a non-issue

discussion about this would also probably be left in https://github.com/NixOS/nixpkgs/issues/315337. until there some real community consensus, nitpicks like this feel a bit silly 🫠